File: Stack/ArrayImplementation/stack.cpp
#include<iostream>
using namespace std;

#define MAX 10

class Stack{
    int top;
  public:
     int arr[MAX];

     Stack(){
         top=-1;
     }
     bool isEmpty(){
         if(top == -1){
             return true;
         }
         return false;
     }
     
     bool isFull(){
         if(top == MAX-1){
             return true;
         }
         return false;
     }

     void push(int data){
         if(isFull()){
             cout<<"Stack Is Full Cant Add More\n";
             return;
         }
         top+=1;
         arr[top]=data;
         return;
     }

     void pop(){
         if(isEmpty()){
             cout<<"Stack is Empty\n";
             return;
         }
         int val=arr[top];
         top-=1;
         cout<<"Poped Value is "<<val<<"\n";
         return;

     }
     void printStack(){
         for(int i=top;i>=0;i--){
             cout<<arr[i]<<" ";
         }
         cout<<"\n";
     }
};

int main(){
   class Stack s;

   s.push(20);
   s.push(30);
   s.push(40);
   s.printStack();
   s.pop();
   s.printStack();

}

File: Stacks/stack.js
class Stack{
    constructor(size){
        this.items=[]
        this.size=size
    }
    isEmpty(){
        return this.items.length==0
    }
    isFull(){
        return this.items.length==this.size
    }
    push(element){
        if(!this.isFull()){
            this.items.push(element)
            console.log(`Item ${element} pushed to stack successfully`)
            return
        }
        console.log("Stack Is Full Cannot add more")
        return   
    }
    pop(){
        if(!this.isEmpty()){
            let element=this.items.pop()
            console.log(`${element} removed successfully`)
        }
        else{
            console.log("Sorry Stack Is Empty")
        }
    }
    peek(){
        if(!this.isEmpty()){
            console.log(this.items[this.items.length-1])
        }
        else{
            console.log("Stack Is Empty")
        }
    }
    printStack(){
        let str=''
        for(let i=0;i<this.items.length;i++){
            str+=this.items[i]+" "
        }
        console.log(str)
    }
    
}

let stack=new Stack(5)
stack.push(10)
stack.push(20)
stack.push(30)
stack.pop()
stack.printStack()